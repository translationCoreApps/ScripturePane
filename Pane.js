
const api = window.ModuleApi;
const React = api.React;
const RB = api.ReactBootstrap;
const {Col, Well, Glyphicon} = RB;
const style = require('./Style');
const VerseDisplay = require('./VerseDisplay');

class Pane extends React.Component {
    constructor() {
        super();
        this.state = {
          chapter: null,
          verse: null
        }
        this.contentStyle = style.pane.content;
        this.headerStyle = style.pane.title;
    }

    componentWillMount(){
      this.getCurrentCheck();
    }
    /**
     * @description This gets the current checks chapter number and verse Number
     * @return {state} chapter # and verse # is saved in the state of the Component
    *******************************************************************************/
    getCurrentCheck() {
      var currentNamespace = api.getCurrentCheckNamespace();
      //TODO: dont forget to remove the temporary dynamic namespaces below
      var currentGroupIndex = api.getDataFromCheckStore('TranslationWordsChecker', 'currentGroupIndex');
      var currentCheckIndex = api.getDataFromCheckStore('TranslationWordsChecker', 'currentCheckIndex');
      var currentCheck = api.getDataFromCheckStore('TranslationWordsChecker', 'groups')[currentGroupIndex]['checks'][currentCheckIndex];
      this.setState({
        chapter: currentCheck.chapter,
        verse: currentCheck.verse
      });
    }

    render() {
      if(this.props.dir == 'ltr'){
        this.contentStyle.direction = 'ltr';
      }else{
        this.contentStyle.direction = 'rtl';
      }
      return (
          <Col md={3} sm={3} xs={3} lg={3}>
              <div style={{float: "right", cursor: "pointer"}} onClick={this.props.remove}>
                <Glyphicon glyph={"remove"} style={{color: "grey"}}/>
              </div>
              <span style={this.headerStyle}>{this.props.heading || ''}</span>
              <div style={this.contentStyle}>
                  <VerseDisplay chapter={this.state.chapter} verse={this.state.verse}
                                input={this.props.content} greek={this.props.greek} />
              </div>
          </Col>
      );
    }
}

module.exports = Pane;
