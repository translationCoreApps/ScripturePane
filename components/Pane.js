
const api = window.ModuleApi;
const React = api.React;
const RB = api.ReactBootstrap;
const {Col, Well, Glyphicon} = RB;
const style = require('../css/Style');
const VerseDisplay = require('./VerseDisplay');
const CloseButton = require('./CloseButton');

class Pane extends React.Component {
    render() {
      let { content, greek, removePane, heading, dir, currentCheck, id, isGatewayLanguage } = this.props;
      let { chapter, verse } = currentCheck;
      let contentStyle;
      if(dir == 'ltr'){
        contentStyle = style.pane.contentLTR;
      }else{
        contentStyle = style.pane.contentRTL;
      }
      return (
          <Col md={3} sm={3} xs={3} lg={3}>
          <div style={style.removePane}
               onClick={() => removePane(id)}>
            <CloseButton />
          </div>
              <span style={style.pane.title}>
                {heading.heading || ''}
              </span>
              <span style={style.headingDescription}>
                {heading.headingDescription || ''}
              </span>

              <div style={contentStyle}>
                  <VerseDisplay chapter={chapter} verse={verse}
                                input={content} greek={greek}
                                isGatewayLanguage = {isGatewayLanguage}
                                phrase={this.props.currentCheck.phrase}
                  />
              </div>
          </Col>
      );
    }
}

module.exports = Pane;
