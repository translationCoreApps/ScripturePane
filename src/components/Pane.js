import React from 'react';
import PropTypes from 'prop-types';
import ContainerDimensions from 'react-container-dimensions';
import {Glyphicon} from 'react-bootstrap';
import style from '../css/Style';
import Verse from './Verse';
import {getTranslation} from "../helpers/localizationHelpers";

class Pane extends React.Component {
  render() {
    let { translate, removePane, index, bibleId, languageId, resourcesReducer: { bibles } } = this.props;
    let { reference } = this.props.contextIdReducer.contextId;
    let {
      direction,
      language_name,
      resource_id,
      description
    } = bibles && bibles[languageId][bibleId] ? bibles[languageId][bibleId]["manifest"] : {};
    direction = direction || 'ltr';
    description = getTranslation(translate, description || "");
    let verseText = bibles && bibles[languageId][bibleId] && bibles[languageId][bibleId][reference.chapter] ? bibles[languageId][bibleId][reference.chapter][reference.verse] : '';
    let languageName = getTranslation(translate, language_name);
    let headingText = bibleId !== "targetBible" ? languageName + " (" + bibleId.toUpperCase() + ")" : (languageName || '');
    let contentStyle;
    const PANECHAR = 9;

    if (direction === 'ltr') {
      contentStyle = style.pane.contentLTR;
    } else {
      contentStyle = style.pane.contentRTL;
    }

    return (
      <div style={index > 0 ? style.otherPane : style.firstPane}>
        <div style={style.verseTitle}>
          <div style={{display: 'flex', flexDirection: 'column', width: '100%'}}>
            <span style={style.pane.title} className={headingText.length > 21 ?
                "hint--bottom hint--medium" : null} aria-label={headingText}>
              {headingText.length > 21 ? headingText.slice(0, 21) + '...' : headingText}
            </span>
            <ContainerDimensions>
              {
                ({width}) =>
                <span style={style.pane.subtitle} className={description.length > width/PANECHAR ?
                    "hint--bottom hint--medium" : null} aria-label={description}>
                  {description.length > width/PANECHAR ?
                      description.slice(0, Math.round(width/PANECHAR)) + "..." :
                      description}
                </span>
              }
            </ContainerDimensions>
          </div>
          <Glyphicon glyph={"remove"} style={{color: "var(--text-color-light)", cursor: 'pointer'}}
                     onClick={() => removePane(index)} title={translate("click_remove_resource")} />
        </div>
        <div style={contentStyle}>
          <Verse
            {...this.props}
            verseText={verseText}
            chapter={reference.chapter}
            verse={reference.verse}
            direction={direction}
            bibleId={resource_id}
            languageId={languageId}
            isCurrent={true}
          />
        </div>
      </div>
    );
  }
}

Pane.propTypes = {
  translate: PropTypes.func.isRequired,
  removePane: PropTypes.func.isRequired,
  index: PropTypes.number.isRequired,
  bibleId: PropTypes.string.isRequired,
  languageId: PropTypes.string.isRequired,
  actions: PropTypes.shape({
    setToolSettings: PropTypes.func.isRequired,
    getWordListForVerse: PropTypes.func.isRequired,
    loadLexiconEntry: PropTypes.func.isRequired,
    showPopover: PropTypes.func.isRequired
  }),
  contextIdReducer: PropTypes.object.isRequired,
  resourcesReducer: PropTypes.object.isRequired

};

export default Pane;
