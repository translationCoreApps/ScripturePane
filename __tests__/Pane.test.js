/* eslint-env jest */

import React from 'react';
import Pane from '../src/components/Pane';

// Tests for AddPanelModal React Component
describe('Test Pane behavior',()=>{
  test('Test book name and language layout.', () => { 
    const props = {
      actions: {},
      contextIdReducer: {
        "contextId":{
                "reference":{
                        "bookId":"mat","chapter":5,"verse":27},
                "tool":"translationWords",
                "groupId":"adultery",
                "quote":"adultery, adulterous, adulterer, adulteress, adulterers, adulteresses",
                "occurrence":1
                
        }
      },
      resourcesReducer: {
        "bibles":{
          "udb":{
            "1":{
              "1":"This is the record of the ancestors of Jesus the Messiah, " + 
                  "the descendant of King David and of Abraham.",
              "2":"Abraham was the father of Isaac. " + 
                  "Isaac was the father of Jacob. Jacob was the father of Judah and of his brothers."
            }
          }
        }
      },
      index: 2,
      bibleId: "udb",
      currentPaneSettings: ["targetLanguage","bhp","udb"]
    };
    const myPane = (
            <Pane {...props} />
    );
    expect(myPane).toMatchSnapshot();
  });
});