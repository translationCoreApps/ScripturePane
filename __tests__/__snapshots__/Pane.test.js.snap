// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Pane behavior Test book name and language layout. 1`] = `
<Pane
  actions={
    Object {
      "getWordListForVerse": [Function],
      "loadLexiconEntry": [Function],
      "setToolSettings": [Function],
      "showPopover": [Function],
    }
  }
  bibleId="udt"
  contextIdReducer={
    Object {
      "contextId": Object {
        "groupId": "adultery",
        "occurrence": 1,
        "quote": "adultery, adulterous, adulterer, adulteress, adulterers, adulteresses",
        "reference": Object {
          "bookId": "mat",
          "chapter": 5,
          "verse": 27,
        },
        "tool": "translationWords",
      },
    }
  }
  currentPaneSettings={
    Array [
      "targetLanguage",
      "bhp",
      "udt",
    ]
  }
  index={2}
  removePane={[Function]}
  resourcesReducer={
    Object {
      "bibles": Object {
        "udt": Object {
          "1": Object {
            "1": "This is the record of the ancestors of Jesus the Messiah, the descendant of King David and of Abraham.",
            "2": "Abraham was the father of Isaac. Isaac was the father of Jacob. Jacob was the father of Judah and of his brothers.",
          },
        },
      },
    }
  }
/>
`;
